packages/safe-tools-client/app/services/wallet.ts

```diff
@@ -84,6 +84,64 @@ export default class Wallet extends Service {
     this.chainConnectionManager.reconnect(this.web3, this.providerId);
   }
 
+  async switchNetwork(chainId: number) {
+    if (this.providerId !== 'metamask') {
+      return;
+    }
+
+    if (!window.ethereum) {
+      return;
+    }
+
+    const chainIdHex = `0x${chainId.toString(16)}`;
+
+    try {
+      //@ts-expect-error currentProvider does not match Web3Provider Property 'request' does not exist on type 'EthereumProvider'
```

Comment:
https://github.com/cardstack/cardstack/pull/3479/files#diff-1cc2eaa548adbb2771ec9077d450bb484fd247fa7a3578c64734d42c70f6110dR12 is suposed to take care of this but the error still remains. I'm researching why.